lib/cli/test/unit/Cardano/CLISpec.hs

```diff
@@ -488,6 +511,40 @@ spec = do
         textRoundtrip $ Proxy @(Port "test")
         textRoundtrip $ Proxy @MnemonicSize
         textRoundtrip $ Proxy @CliWalletStyle
+        textRoundtrip $ Proxy @DerivationIndex
+        textRoundtrip $ Proxy @DerivationPath
+
+    describe "DerivationPath/Index fromText goldens" $ do
+        fromTextGolden @DerivationIndex "" "should fail" $
+            Left "An empty string is not a derivation index!"
+
+        fromTextGolden @DerivationIndex "0'" "should fail" $
+            Left "\"0'\" is not a number."
+
+        fromTextGolden @DerivationIndex "4294967295H" "should fail" $
+            Left "6442450943 is too high to be a derivation index."
+
+        fromTextGolden @DerivationPath "" "should fail" $
+            Left "An empty string is not a derivation index!"
+
+        fromTextGolden @DerivationPath "44H/0H/0" "" $
+            Right . DerivationPath . map DerivationIndex $
+                [0x80000000 + 44, 0x80000000 + 0, 0]
```

Comment:
maybe we can give name to `0x8000000` - it is minBound of soft derivation index, right?

Reply:
It's the minBound of the **hardened** indexes. Can move to a named let-binding just here

(wanted to duplicate the definition at https://github.com/input-output-hk/cardano-wallet/blob/c0479c0258bd6a8e8fd00d5e708a5fbe2e3fcfba/lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs#L230 on purpose though)

Reply:
*hardened* yes yes